paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -42,18 +46,33 @@ import java.util.Locale
 class PaparazziPlugin : Plugin<Project> {
   @OptIn(ExperimentalStdlibApi::class)
   override fun apply(project: Project) {
-    require(project.plugins.hasPlugin("com.android.library")) {
-      "The Android Gradle library plugin must be applied before the Paparazzi plugin."
+    project.plugins.withId("com.android.library") {
+      setup(project, project.extensions.getByType(LibraryExtension::class.java).libraryVariants)
     }
+    project.plugins.withId("com.android.application") {
+      error(
+        "The Android Gradle Plugin must be for a library module. " +
+                "See https://github.com/cashapp/paparazzi/issues/107."
+      )
+    }
+    project.afterEvaluate {
+      try {
+        // Use named instead of findByName to make sure the task doesn't get created unnecessarily.
+        it.tasks.named("verifyPaparazzi")
+      } catch(ex: UnknownTaskException) {
+        error("The Android Gradle library plugin must be applied together with Paparazzi plugin.")
+      }
+    }
+  }
 
+  private fun <T> setup(project: Project, variants: DomainObjectSet<T>)
+      where T : BaseVariant, T : TestedVariant {
```

Comment:
```suggestion
  private fun setupPaparazzi(project: Project) {
```

Reply:
Yeah, historical, left it because it's non-trivial, but I guess yagni until #107 can be resolved, which we can't see yet.